### PR TITLE
Help window: show progress bar in status bar only when loading a page

### PR DESCRIPTION
**Pull request by howlger**
Problem:
In the _Help > Help Contents_ window in the status bar, sometimes the progress bar is missed to be hidden, even after page loading is completed. Also, JavaScript requests can cause a (stale) progress bar to be shown.

This happens in both, the legacy Help UI, and more often in the  [_Modernized Eclipse Help UI prototype_](https://github.com/eclipse-platform/eclipse.platform.ua/blob/58e2060a6efc181f16cef9b1d7224460ac65cb2c/org.eclipse.help.webapp/m/README.md).

See also @iloveeclipse's [comment on the _Modernized Eclipse Help UI prototype_](https://git.eclipse.org/r/c/platform/eclipse.platform.ua/+/170090/5#message-758e82ce5ab8347b797a36a3b5b0e2b25c95a90e):

> One more bug need to be addressed: the status bar on the bottom never disappears after indexing or search is done

### CODE REVIEW

File: org.eclipse.help.ui/src/org/eclipse/help/ui/internal/browser/embedded/EmbeddedBrowser.java [hunk 1]
```diff
@@ -304,31 +305,50 @@ public void changing(LocationEvent e) {
 	}
 
 	private void initializeStatusBar(Browser browser) {
+
+		// Text
 		browser.addStatusTextListener(event -> {
-			event.text = event.text.replaceAll("&", "&&"); //$NON-NLS-1$ //$NON-NLS-2$
+			event.text = event.text.replace("&", "&&"); //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (mickaelistria) [inline org.eclipse.help.ui/src/org/eclipse/help/ui/internal/browser/embedded/EmbeddedBrowser.java:311]: Seems like a bugfix, isn't it worth a dedicated PR?

Comment (howlger) [reply org.eclipse.help.ui/src/org/eclipse/help/ui/internal/browser/embedded/EmbeddedBrowser.java:311]: That's just a (performance) clean-up, not a bugfix.

Comment (howlger) [reply org.eclipse.help.ui/src/org/eclipse/help/ui/internal/browser/embedded/EmbeddedBrowser.java]: I see and I have reverted the clean-up.

File: org.eclipse.help.ui/src/org/eclipse/help/ui/internal/browser/embedded/EmbeddedBrowser.java [hunk 2]
```diff
@@ -304,31 +305,50 @@ public void changing(LocationEvent e) {
...
+			public void changing(LocationEvent event) {
+				if (event.location == null || event.location.startsWith("javascript:")) //$NON-NLS-1$
+					return;
+				changingLocationHolder.set(event.location);
 			}
 
 			@Override
-			public void completed(ProgressEvent event) {
-				statusBarSeparator.setVisible(false);
-				statusBarProgress.setVisible(false);
```
Comment (mickaelistria) [inline org.eclipse.help.ui/src/org/eclipse/help/ui/internal/browser/embedded/EmbeddedBrowser.java:329]: Wouldn't it be worth keeping this method as a fallback strategy?

**Comment by mickaelistria** [comment]
OK, thanks.

**Pull request merged**